odoo/addons/base/transpiler/transpiler_js.py

```diff
@@ -0,0 +1,393 @@
+import re
+
+
+class TranspilerJS:
```

Comment:
Awkward naming, IMHO.  Why not `JSTranspiler`?
It's like calling a method `fields_get` instead of `get_fields`.  Oh wait...  Ahem :thinking:

Reply:
By the way, why a class?
```python
transpiler = TranspilerJS(content, url)
result = transpiler.convert()
```
Seriously, this should be a function.

Reply:
TBF the class does serve as a convenient holder for the parameters, and once the class is there using a method does make sense.

But that class could be an internal detail of the thing with a toplevel function providing a simple / convenient API, similar to what Armin Ronacher tends to do.

And on a simplicity / convenience basis, maybe the URL should be passed before the content? Seems more intuitive to me that if the caller provides both, the "pointer" would be specified before the "pointee".